AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -971,3 +984,51 @@ - (void)dealloc
 }
 
 @end
+
+@implementation ASMainQueueSerialQueue
```

Comment:
Since this code is completely generic in its application - although I am afraid to say I hope we never need to use it again, LOL - what do you think about moving it to ASThread or a similar helper / utility file?

Some solid magic in the runBlocks method...